include/vast/Dialect/HighLevel/HighLevelOps.td

```diff
@@ -986,7 +986,26 @@ class TypePreservingUnOp< string mnemonic, list< Trait > traits = [] >
 def PlusOp  : TypePreservingUnOp< "plus", [] >;
 def MinusOp : TypePreservingUnOp< "minus", [] >;
 def NotOp   : TypePreservingUnOp< "not", [] >;
-def LNotOp  : TypePreservingUnOp< "lnot", [] >;
+
+class LogicalUnOp< string mnemonic, list< Trait > traits = [] >
+    : HighLevel_Op< mnemonic, !listconcat(traits, [NoSideEffect]) >
+    , Arguments< (ins AnyType:$arg) >
```

Comment:
I did not check the standard but isn't this also required to be int or bool?

Reply:
C++ does require conversion of the operand to bool but the C standard only says that the operand is a scalar type :/